Update design for listings directory table #173346911

james-wills-sf opened the pull request:
Ticket [#173346911](https://www.pivotaltracker.com/n/projects/1405352/stories/173346911)

### Screenshot: rental desktop view
<img src="https://user-images.githubusercontent.com/64036574/88854558-4f187280-d1a6-11ea-80fb-b155383175ac.png" width="630" />

### Screenshot: sale desktop view
<img src="https://user-images.githubusercontent.com/64036574/88607419-0ccf2400-d034-11ea-8eae-b3711b5ece13.png" width="630" />

### Screenshot: rental mobile view
<img src="https://user-images.githubusercontent.com/64036574/88854532-44f67400-d1a6-11ea-8771-fb53008e39cc.png" width="430" />

### Screenshot: sale mobile view
<img src="https://user-images.githubusercontent.com/64036574/88854616-66576000-d1a6-11ea-8bbe-e3184f063dbb.png" width="430" />

### Screenshot: sale desktop view -- very long text edge case
<img src="https://user-images.githubusercontent.com/64036574/88854567-550e5380-d1a6-11ea-96e2-df1d85ad4503.png" width="530" />


todo:
- [ ] remove the mock data
- [x] fix sale listings mobile view
- [x] fix rental listings mobile view
- [x] tests
- [x] a11y pass

akegan commented:
@cpalmatier we have a question on how to make these changes work for mobile view:

## What we have currently
for rental:
![image](https://user-images.githubusercontent.com/11825994/88712529-b95de400-d0ce-11ea-9cac-e2c580200bae.png)
for sale:
![image](https://user-images.githubusercontent.com/11825994/88712713-0346ca00-d0cf-11ea-9fef-b1a309cd1606.png)

## What it looks like just making the changes in the mock-ups
for rental:
 ![image](https://user-images.githubusercontent.com/11825994/88712566-c549a600-d0ce-11ea-876b-0f479be19a2f.png)
and sale:
![image](https://user-images.githubusercontent.com/11825994/88712677-f32eea80-d0ce-11ea-913c-9435ef31f4af.png)

## The issues
The text overlaps on mobile view for sales and on rental, the 1BR plus availability is a bit cramped. 

## Potential solutions
1. Maybe it would look better if we go back to having the small text (like per month) go below the values? 
1. Or maybe we have to drop some of the info on mobile view or on the directory view for now, until we can revisit?
1. James has offered to break these tables up into separate tables for mobile and desktop view so we don't have to rely on css magic to make the mobile tables. We could do that in the scope of this ticket if the changes are small, or tackle a bigger refactor when we tackle the mobile view for the listing details tables?

cpalmatier commented:
@akegan @james-wills-sf I think the idea of separating out mobile and desktop components for pricing tables is one whose time has come! Let's do it.

Given separate mobile components here, I think the idea of "break[ing] the HOA dues onto their own line in addition to the 'Sales price' line" is the correct solution. I would start with that, and then consider pulling out some info from the mobile view if we're still having trouble fitting everything in a reasonable way (lmk if we need to do that, and I can offer suggestions about what to pull out). I'm also fine with moving the small text onto a 2nd line (as it was before) but _only_ if there's no alternative—keeping the label and value text on the same baseline makes it a little easier to parse, and these views are complex enough that every little bit helps in terms of readability.

james-wills-sf commented:
@cpalmatier I just updated the screenshots in the PR description with the new mobile views. I moved availability and HOA to their own lines. Does this seem okay to you? We can always skip lines entirely for mobile if we feel it's too long

james-wills-sf commented:
@akegan I went ahead and marked this PR as ready for review. It's still using mock data but everything else is done. Up to you if you want to wait to review til after the salesforce change is in, but I don;t anticipate anything but the mock code to change.

akegan commented on app/assets/javascripts/listings/components/rental-stats.html.slim:
```diff
@@ -1,11 +1,33 @@
+
+    tbody ng-include="'listings/templates/property-card/_rental-stats-row-mobile.html'" ng-init="summary = $ctrl.listing.unitSummaries.reserved"
+
+/ Destop table
```
```suggestion
/ Desktop table
```
minor typo

akegan commented on app/assets/javascripts/listings/components/sale-stats.html.slim:
```diff
@@ -1,12 +1,35 @@
+
+    tbody ng-include="'listings/templates/property-card/_sale-stats-row-mobile.html'" ng-init="summary = $ctrl.listing.unitSummaries.reserved"
+
+/ Destop table
```
```suggestion
/ Desktop table
```
same typo

akegan commented on app/assets/javascripts/listings/templates/property-card/_rental-stats-row.html.slim:
```diff
@@ -1,15 +1,17 @@
 tr ng-repeat-start="unit_summary in ::summary" ng-if="::unit_summary.minMonthlyRent || unit_summary.minPercentIncome"
-  td.is-subtitled data-th="{{'t.unit_type' | translate}}"
+  td.is-subtitled data-th="{{'t.units' | translate}}"
```
should we rename these files to -desktop just as a hint that there's another file out there?

james-wills-sf commented:
@akegan updated this to have the new sales table layout, and to handle null absoluteMinIncome values for rentals. Even though there is still the salesforce bug for AMI tiers listings, I still think this is ready for re-review.

### Desktop sales
<img width="625" alt="- 2020-08-12 at 1 47 46 PM" src="https://user-images.githubusercontent.com/64036574/90066747-016e3080-dca3-11ea-9951-aeaccafa2b09.png">

### 320px sales
<img width="365" alt="- 2020-08-12 at 1 48 08 PM" src="https://user-images.githubusercontent.com/64036574/90066807-14810080-dca3-11ea-99ed-be9a07fc7fbb.png">

### Tagalog desktop sales
<img width="636" alt="- 2020-08-12 at 1 50 43 PM" src="https://user-images.githubusercontent.com/64036574/90066831-1ea2ff00-dca3-11ea-976f-84ff808acdf2.png">

### Tagalog 320px sales
Note that there is some text cutoff, we decided in the related pattern library PR that we're not going to fix this for now.
<img width="303" alt="- 2020-08-12 at 1 51 06 PM" src="https://user-images.githubusercontent.com/64036574/90066870-2bbfee00-dca3-11ea-9b87-8629a4802a7f.png">

### Rental with $0 minimums
Technically this is still a bug, these particular minimums shouldn't be $0 and are currently being fixed on the backend, but they show what this looks like when absoluteMinIncome is null.
<img width="616" alt="- 2020-08-12 at 1 55 49 PM" src="https://user-images.githubusercontent.com/64036574/90067174-8d805800-dca3-11ea-8735-cfd7832b6afa.png">

james-wills-sf commented:
@akegan sounds good. I'll look into the missing rent.

Missing income range is expected, there are only 3 populated rental listings on full for us, a0W0P00000HbryHUAR, a0W0P00000HE9sfUAD, anda0W1F000002DHG3UAO, so I wouldn't expect Alexander to have the right income info. It should have rent info though...

akegan commented:
@james-wills-sf can we add alexander residence to the listings Larry is testing with so we have one of the % income listings as a test? That's the only case where we'd expect there to be no min income for rentals so would be nice to check that case.

akegan commented:
@james-wills-sf income ranges on listing directory look great. Looks like rent is missing on alexander residence still

james-wills-sf commented:
@akegan looks like I caused a regression with units where rent is calculated by % income. This is fixed now:

![- 2020-08-17 at 12 07 58 PM](https://user-images.githubusercontent.com/64036574/90434534-6ef2d600-e082-11ea-8699-a6f84fe2b97e.png)

![- 2020-08-17 at 12 08 16 PM](https://user-images.githubusercontent.com/64036574/90434601-8c27a480-e082-11ea-897e-b1856a296a06.png)
